test/blackbox-tests/test-cases/dynamic-dependencies/read-macro-produces-dyn-deps.t

```diff
@@ -0,0 +1,41 @@
+Tests for dynamic dependencies computed from the `%{read:...}` family of macros
+
+  $ cat > dune-project <<EOF
+  > (lang dune 3.0)
+  > EOF
+
+Define 2 rules and a file containing their paths
+
+  $ cat > deps.d <<EOF
+  > depA
+  > depB
+  > EOF
+
+  $ cat > dune <<EOF
+  > (rule
+  >  (target depA)
+  >  (action
+  >   (bash "echo depA > %{target}")))
```

Comment:
I suggest to pick different strings for file contents (e.g. `contentsA` for `depA`) to avoid a possible confusion between the names and contents.